neuron/src/lib/Neuron/Zettelkasten/Query.hs

```diff
@@ -49,6 +50,12 @@ runZettelQuery zs = \case
 runGraphQuery :: ZettelGraph -> GraphQuery r -> r
 runGraphQuery g = \case
   GraphQuery_Id -> g
+  GraphQuery_BacklinksOf conn zid ->
+    case getZettel zid g of
+      Nothing -> []
+      Just z2 ->
+        mapMaybe (\z1 -> (,z1) <$> getConnection z1 z2 g) $
```

Comment:
We shouldn't have to lookup the connection a *second* time, if `backlinks` can return it along with the zettel (i.e., `[(Connection, Zettel)]` instead `[Zettel]`).

Reply:
@srid Do you have a suggestion for how `backlinks` could return the connection along with the Zettel?  It looks like I'd have to replace `preSet` with something that doesn't discard the graph edges.  I don't see a function that can do what `preSet` does yet preserve the edges.

Would I need to generate a `Forest` and then fold the forest into a list of vertices and edges?

Reply:
Note that in latest master, backlinks uses `preSetWithEdgeLabelMany` which in turn uses `postSet` (for efficiency; see #245). And `postSet` is defined in the alga library as `Map.findWithDefault Set.empty x . adjacencyMap` - i.e., it just looks up the map (which was produced from the 'inverse' graph). 

You could probably instead do `Map.lookup x . adjacencyMap` which gets you a `Map a e` (i.e., backlinks with connection).

Reply:
That sounds good.  I'll rebase and look around.  Thank you.

Reply:
The `backlinks` function now returns the Zettel *and* connection.